models/solana_utils/solana_utils_daily_balances.sql

```diff
@@ -6,50 +6,48 @@
         materialized='incremental',
         file_format = 'delta',
         incremental_strategy='merge',
+        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.day')],
         unique_key = ['token_mint_address', 'address','day'],
         post_hook='{{ expose_spells(\'["solana"]\',
                                     "sector",
                                     "solana_utils",
                                     \'["ilemi"]\') }}')
 }}
 
-WITH 
+WITH
       tokens_accounts as (
-            SELECT 
+            SELECT
             distinct address
-            FROM {{ source('solana','account_activity') }}
-            WHERE tx_success
-            AND token_mint_address is not null
+            FROM {{ ref('solana_utils_token_accounts') }}
       )
-      
       , updated_balances as (
             SELECT
-                  address 
+                  address
                   , date_trunc('day', block_time) as day
                   , token_mint_address
-                  , cast(post_balance as double)/1e9 as sol_balance --lamport -> sol 
+                  , cast(post_balance as double)/1e9 as sol_balance --lamport -> sol
                   , post_token_balance as token_balance --tokens are already correct decimals in this table
                   , token_balance_owner
                   , row_number() OVER (partition by address, date_trunc('day', block_time) order by block_slot desc) as latest_balance
             FROM {{ source('solana','account_activity') }}
-            WHERE tx_success 
+            WHERE tx_success
             AND (
                   --if the address is a token_mint_account, then mint_address should not be null
                   (address NOT in (select address from tokens_accounts))
                   or (address in (select address from tokens_accounts) AND token_mint_address is not null)
                   )
             {% if is_incremental() %}
-            AND block_time >= date_trunc('day', now() - interval '1' day)
+            AND {{incremental_predicate('block_time')}}
```

Comment:
Incremental macro might not actually be a good idea here because if we go for hours instead of days it would mess things up here.